File: backend/ArhiTodo.Domain/Entities/DTOs/CardGetDto.cs
namespace ArhiTodo.Domain.Entities.DTOs;

public class CardGetDto
{
    public int CardId { get; set; }
    
    public required string Position { get; set; }
    
    public bool IsDone { get; set; }
    
    public required string CardName { get; set; }

    public string CardDescription { get; set; } = string.Empty;
    
    public List<int> LabelIds { get; set; } = [];
    public List<ChecklistGetDto> Checklists { get; set; } = [];
}

File: frontend/src/Models/BackendDtos/Kanban/CardGetDto.ts
import type {ChecklistGetDto} from "./ChecklistGetDto.ts";

export type CardGetDto = {
    cardId: number;
    position?: string;
    cardName: string;
    cardDescription: string;
    isDone: boolean;
    cardUrgencyLevel: number;
    labelIds: number[];
    assignedUserIds: string[];
    checklists: ChecklistGetDto[];
}